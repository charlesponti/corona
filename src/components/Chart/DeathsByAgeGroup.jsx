import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Grid from '@material-ui/core/Grid'

import cx from 'classnames'
import styles from './DeathsByAgeGroup.module.css'

import deathsByAgeGroup from '../../data/deaths.json'
import { CardHeader } from '@material-ui/core'

function toNumber(string) {
  return Number(string.replace(/,/ig, ''))
}

function getPercentageOfDeaths(row) {
  return (toNumber(row["All Deaths involving COVID-19"]) / toNumber(row["Deaths from All Causes"]) * 100).toFixed(2)
}

export default function DeathsByAgeGroup () {
  return (
    <Grid container justify="center" className={cx(styles.container)}>
      <Grid item component={Card} xs={12} sm={10}>
        <CardContent>
          <CardHeader title="Deaths by Covid"/>
          <Grid container>
            {deathsByAgeGroup.map((row) => {
              const ageGroup = row["Age group"].toLowerCase()
              return (
                <Grid container xs={12} justify="center" key={row["Age group"]}>
                  <Grid item xs={6} sm={3}>
                    {ageGroup}
                  </Grid>
                  <Grid item xs={6} sm={3}>
                    {getPercentageOfDeaths(row)}%
                  </Grid>
                </Grid>
            )})}
          </Grid>
        </CardContent>
      </Grid>
    </Grid>
  )
}

DeathsByAgeGroup.propTypes = {
  countryCode: PropTypes.string
}
