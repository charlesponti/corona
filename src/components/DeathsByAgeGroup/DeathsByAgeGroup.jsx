import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Grid from '@material-ui/core/Grid'

import styles from './DeathsByAgeGroup.module.css'
import theme from '../../theme.module.css'

import deathsByAgeGroup from '../../data/deaths.json'
// import { getPercentageOfDeaths } from '../../utils'

export default function DeathsByAgeGroup () {
  return (
    <Grid container justify="center" className={styles.container}>
      <Grid item component={Card} xs={11} className={theme.card}>
        <h2 style={{ paddingLeft: '16px' }}>Deaths by Age Group</h2>
        <CardContent>
          <div className={styles.data}>
            {deathsByAgeGroup.map((row) => {
              const ageGroup = row["Age group"].toLowerCase()
              return (
                <Grid container key={row["Age group"]} className={styles.ageGroup}>
                  <Grid item xs={10} sm={7}>
                    {ageGroup.replace('years', '').replace('year', '')}
                  </Grid>
                  <Grid item xs={2} sm={3} className={styles.deaths}>
                    {row["All Deaths involving COVID-19"]}
                  </Grid>
                  {/* <Grid item xs={2} sm={3}>
                    {getPercentageOfDeaths(row)}%
                  </Grid> */}
                </Grid>
            )})}
          </div>
        </CardContent>
      </Grid>
    </Grid>
  )
}

DeathsByAgeGroup.propTypes = {
  countryCode: PropTypes.string
}
