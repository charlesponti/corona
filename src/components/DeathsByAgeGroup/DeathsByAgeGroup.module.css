.container {
  margin: 2rem 0;
  max-width: 700px;
}

.ageGroup {
  margin-bottom: 0.25rem;
}

.deaths {
  color: #ff5a5a;
}