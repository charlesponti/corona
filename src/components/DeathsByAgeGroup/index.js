export { default as DeathsByAgeGroup } from './DeathsByAgeGroup'