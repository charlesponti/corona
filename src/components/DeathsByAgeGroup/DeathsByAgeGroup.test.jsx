import React from 'react'
import { render } from '@testing-library/react'
import DeathsByAgeGroup from './DeathsByAgeGroup'

test('should render DeathsByAgeGroup', () => {
  const { container } = render(
    <DeathsByAgeGroup />
  )
  expect(container).toMatchSnapshot()
})
