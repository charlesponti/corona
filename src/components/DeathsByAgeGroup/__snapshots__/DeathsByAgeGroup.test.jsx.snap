// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeathsByAgeGroup 1`] = `
<div>
  <div
    class="MuiGrid-root container MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiGrid-root card MuiGrid-item MuiGrid-grid-xs-11 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h2
        style="padding-left: 16px;"
      >
        Deaths by Age Group
      </h2>
      <div
        class="MuiCardContent-root"
      >
        <div
          class="data"
        >
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              all ages
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              217,348
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              under 1 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              25
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              1–4 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              15
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              5–14 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              39
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              15–24 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              400
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              25–34 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              1,677
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              35–44 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              4,333
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              45–54 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              11,476
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              55–64 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              27,569
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              65–74 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              46,988
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              75–84 
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              57,866
            </div>
          </div>
          <div
            class="MuiGrid-root ageGroup MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
            >
              85  and over
            </div>
            <div
              class="MuiGrid-root deaths MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
            >
              66,960
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
